src/common/common.h

```diff
@@ -152,6 +152,11 @@ UniqueID globally_unique_id(void);
 
 typedef UniqueID ObjectID;
 
+#ifdef __cplusplus
+size_t hashObjectID(const ObjectID &key);
+bool operator==(const ObjectID& x, const ObjectID& y);
```

Comment:
I tried all combinations of inlining and the speed of `ray.wait` on 10^5 objects is unaffected (120ms on my laptop). We can analyze this more when we put together the benchmarks.